### PR TITLE
SC-5710 courses a11y

### PR DESCRIPTION
**Pull request by pawelchoniawko**
# Description
A11y courses

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-5710

### CODE REVIEW

**Comment by qnowak** [comment]
This button is not accessible from the keyboard
image

**Comment by qnowak** [comment]
This button is not reachable from the keyboard
image

**Comment by qnowak** [comment]
Missing form label in form-import-course.hbs
image

**Comment by qnowak** [comment]
Missing text in de.json 
image

File: views/courses/create-course.hbs [hunk 1]
```diff
@@ -148,7 +148,7 @@
                           {{#each course.times}}
                               <tr class="course-time" id="{{this._id}}">
 								<td>
-									<button class="btn-link course-time-delete" type="button" data-id="{{this._id}}"
+									<button aria-label="delete course" class="btn-link course-time-delete" type="button" data-id="{{this._id}}"
```
Comment (qnowak) [inline views/courses/create-course.hbs:151]: Add aria-label text to i18n

File: views/lib/error_content.hbs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div class="error-header-wrapper pb-3">
-    <img class="error-header-image" src="/images/broken-pen.svg" alt="{{$t "global.img_alt.brokenPen"}}" />
+    <img class="error-header-image" src="/images/broken-pen.svg" alt="" />
```
Comment (qnowak) [inline views/lib/error_content.hbs:2]: Add role="presentation"

File: views/lib/csrf_content.hbs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div class="error-header-wrapper pb-3">
-    <img class="error-header-image" src="/images/broken-pen.svg" alt="{{$t "global.img_alt.brokenPen"}}" />
+    <img class="error-header-image" src="/images/broken-pen.svg" alt="" />
```
Comment (qnowak) [inline views/lib/csrf_content.hbs:2]: Add role="presentation"

File: views/courses/components/groups.hbs [hunk 4]
```diff
@@ -43,6 +43,7 @@
 			{{#unless @root.isArchived}}
 				<div class="add-button">
 					<a href="/courses/{{../id}}/groups/add"
+						aria-label="{{$t "courses._course.groups.text.addGroup"}}"
 						class="btn btn-primary {{#ifneq 0 (arrayLength ../courseGroups)}}w-100{{else}}btn-empty{{/ifneq}}"
 						title="{{$t "courses._course.groups.text.addGroup"}}">
```
Comment (qnowak) [inline views/courses/components/groups.hbs:48]: Do we need the title here when you added aria-label ?

File: views/courses/components/tasks.hbs [hunk 5]
```diff
@@ -104,6 +104,7 @@
 		{{#userHasPermission "COURSE_EDIT" }}
 			{{#unless @root.isArchived}}
 				<a href="/homework/new?course={{../_id}}"
+					aria-label="{{$t "global.button.addTask"}}"
 					class="btn btn-primary {{#if ../assignedHomeworks}}w-100{{else}}btn-empty{{/if}}"
 					title="{{$t "global.button.addTask"}}">
```
Comment (qnowak) [inline views/courses/components/tasks.hbs:109]: Do we need the title here when you added aria-label ?

File: views/courses/components/topics.hbs [hunk 6]
```diff
@@ -62,12 +62,14 @@
 								</a>
 								<a href="/courses/{{../courseId}}/topics/{{../_id}}/edit"
 									class="btn btn-secondary btn-sm btn-icon-card"
+									aria-label="{{$t "global.button.editTopic"}}"
 									title="{{$t "global.button.editTopic"}}">
```
Comment (qnowak) [inline views/courses/components/topics.hbs:66]: Same

File: views/courses/components/topics.hbs [hunk 7]
```diff
@@ -62,12 +62,14 @@
...
 									<i class="fa fa-pencil"></i>
 								</a>
 								{{/unless}}
 								<a href="#" data-href="{{../_id}}" data-courseId="{{../courseId}}"
 									class="btn btn-secondary btn-sm btn-create-share btn-icon-card"
+									aria-label="{{$t "courses._course.topic.text.shareTopic"}}"
 									title="{{$t "courses._course.topic.text.shareTopic"}}">
```
Comment (qnowak) [inline views/courses/components/topics.hbs:73]: Same

File: views/courses/components/topics.hbs [hunk 8]
```diff
@@ -144,6 +146,7 @@
 		<div class="add-button">
 			<a href="/courses/{{../_id}}/topics/add"
 				class="btn btn-add {{#if ../lessons}}btn-secondary w-100{{else}}btn-primary btn-empty{{/if}} center"
+				aria-label="{{$t "courses._course.topic.text.addTopic"}}"
 				title="{{$t "courses._course.topic.text.addTopic"}}">
```
Comment (qnowak) [inline views/courses/components/topics.hbs:150]: Same

File: views/courses/components/topics.hbs [hunk 9]
```diff
@@ -257,6 +260,7 @@
 		<div class="add-button">
 			<a href="/courses/{{@root._id}}/topics/add/neweditor"
 				class="btn btn-add {{#if @root.newLessons}}btn-secondary w-100{{else}}btn-primary btn-empty{{/if}} center"
+				aria-label="{{$t "courses._course.topic.text.addTopicWithNewEditor"}}"
 				title="{{$t "courses._course.topic.text.addTopicWithNewEditor"}}">
```
Comment (qnowak) [inline views/courses/components/topics.hbs:264]: Same

**Review by qnowak — CHANGES_REQUESTED** [review]

**Review by qnowak — DISMISSED** [review]
image
This still need to be done

File: views/courses/edit-course.hbs [hunk 10]
```diff
@@ -120,7 +120,7 @@
 										{{#ifeq @index 0}}{{#unless ../../../course.color}}checked="checked"
 										{{/unless}}{{/ifeq}} {{#if @root.course.isArchived}}disabled{{/if}} />
 									<label for="input{{../this}}" class="color-picker__color"
-										style="background: {{../this}}"></label>
+										style="background: {{../this}}">background: {{../this}}</label>
```
Comment (qnowak) [inline views/courses/edit-course.hbs:123]: Is that correct ?

File: views/courses/create-course.hbs [hunk 11]
```diff
@@ -68,14 +68,14 @@
...
-                        <label for="input0" class="color-picker__color" style="background: {{course.color}}"></label>
+                        <label for="input0" class="color-picker__color" style="background: {{course.color}}">{{course.color}}</label>
                       </div>
                     {{/if}}
                     {{#each colors}}
                       {{#ifneq this ../course.color}}
                         <div class="color-picker__item">
                           <input id="input{{../this}}" type="radio" class="color-picker__input" name="color" value="{{../this}}" {{#ifeq @index 0}}{{#unless ../../../course.color}}checked="checked"{{/unless}}{{/ifeq}}/>
-                          <label for="input{{../this}}" class="color-picker__color" style="background: {{../this}}"></label>
+                          <label for="input{{../this}}" class="color-picker__color" style="background: {{../this}}">input{{../this}}</label>
```
Comment (qnowak) [inline views/courses/create-course.hbs:78]: Is that `input{{../this}}` necessary?

File: views/courses/components/topics.hbs (hunk 7 above)
Comment (haniapalac) [reply views/courses/components/topics.hbs:73]: I think aria-label or a title is not needed in places where there is a visual label in form of a span, you can remove them.

Comment (haniapalac) [reply views/courses/components/topics.hbs:73]: Check for all such places in the code.

**Comment by haniapalac** [comment]
there are a couple of things I found:
image
the order of the tabbing of the buttons is not logical, 'weiter' is the last to tab and should be the first 

image
You can remove the aria-label here because there is a visual label (span). then it will read also the number of courses

image
this edit page needs adding a11y (associating labels with inputs e.g.)

image
The 'Kurs teilen' button is not tabbable

File: views/lib/default.hbs [hunk 12]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html lang="{{userLanguage}}" timezone="{{currentTimezone}}">
+<html lang="{{#if userLanguage}} {{userLanguage}} {{else}}de{{/if}}" timezone="{{currentTimezone}}">
```
Comment (ABialoskorski) [inline views/lib/default.hbs:2]: I think this is not necessary here since language has default as 'de'

Comment (haniapalac) [reply views/lib/default.hbs:2]: I've noticed userLanguage is undefined.

Comment (ABialoskorski) [reply views/lib/default.hbs:2]: okay ;)

**Review by ABialoskorski — APPROVED** [review]

**Review by haniapalac — APPROVED** [review]

**Pull request merged**